Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-FIPFSScanEngineVersionState.ps1

```diff
@@ -58,12 +61,17 @@ Function Get-FIPFSScanEngineVersionState {
                     -CatchActionFunction ${Function:Invoke-CatchActions}
 
                 if ($null -ne $scanEngineVersions) {
-                    [Int64]$highestScanEngineVersion = ($scanEngineVersions.Name | Measure-Object -Maximum).Maximum
+                    if ($scanEngineVersions.Failed) {
+                        Write-Verbose "Failed to find the scan engine directory"
+                        return $null
+                    } else {
+                        [Int64]$highestScanEngineVersion = ($scanEngineVersions.Name | Measure-Object -Maximum).Maximum
+                    }
                 } else {
                     Write-Verbose "No FIP-FS scan engine version(s) detected"
```

Comment:
Changes made to the check are looking good to me. One thing to point out: 
I think we should change line 71 by either removing the `else` statement or changing the wording and adding `return $null`. 
We should only end up here if `GetHighestScanEngineVersionNumber` returns `$null`. This can be the case if, a file name + extension was passed to the `GetHighestScanEngineVersionNumber` method (we have a RegEx filter in place to check whether `$ExchangeSubDir` is a directory path or not and if it's not we simply return `$null`. The other case would be a catastrophic failure of the call and nothing's returned.